client/src/components/Clan/RoleAssignmentModal.tsx

```diff
@@ -0,0 +1,147 @@
+import { Box, Checkbox, FormControl, InputLabel, MenuItem, Paper, Select } from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import FantasyButton from '../FantasyButton';
+import Text from '../Text';
+import { useAlert } from '../../hooks/useAlert';
+import Server from '../../utils/Server';
+import catchError from '../../utils/catchError';
+
+interface Member {
+  id: string;
+  name: string;
+}
+
+interface Role {
+  id: string;
+  name: string;
+}
+
+interface RoleAssignmentModalProps {
+  open: boolean;
+  onClose: () => void;
+  clanId: string;
+  role: Role | null;
+  members: Member[];
+  onAssigned: () => void;
+}
```

Comment:
I would prefer having types instead of interfaces.
I know the project still has a lot of interfaces so I understand why you used them, but I recently stopped using them entirely, be it professionally or in personal projects, as I find their use dangerous.
Let's say we import an interface from another file:
```
import Test from 'test';
```
If we declare another interface with the same name, forgetting this interface already exists, we end up merging both declarations, instead of seeing a TS error:
```ts
interface Test {
	a: number;
}
interface Test {
	b: string;
}

// Will result in:

interface Test {
	a: number;
	b: string;
}
```

So yeah, TLDR: use types instead of interfaces for safety